spec/features/stories_manage_spec.rb

```diff
@@ -113,6 +113,7 @@
 
     DESC
 
+    sleep(1)
     fill_in "story[description]", with: desc
```

Comment:
@fbuys if you may me, I don't think adding the sleep statement it's better to add the javascript listener, it will slow down the test suit at some point.

Reply:
That sounds reasonable @JuanVqz , however it still fails locally even with the javascript listener. Do you know if I can add a sleep for less than 1 second?

Reply:
I'm not pretty sure but I guess yes `sleep(0.5)` may work.

I think what about if we show a loading icon on the page when the user is writing, we can add an expectation the loading icon should not be there when the preview gets updated.

Reply:
I'll share that I try to avoid `sleep` calls in test if at all possible. I'll try to have the test look for a DOM change that will indicate it's ok for the test to proceed. If for any reason that's not possible or isn't reliable, this is a module I used in a past project. It's from jQuery days... It looks like it could be adapted for non-jQuery use with this: https://stackoverflow.com/a/287212/595962

```
module WaitForAjax
  def wait_for_ajax
    Timeout.timeout(Capybara.default_max_wait_time) do
      loop until finished_all_ajax_requests?
      yield
    end
  end

  def finished_all_ajax_requests?
    page.evaluate_script('jQuery.active').zero?
  end
end

RSpec.configure do |config|
  config.include WaitForAjax, type: :feature
end
```

To give credit where it's do, I got the idea from here https://thoughtbot.com/blog/automatically-wait-for-ajax-with-capybara

I added the yield so you can use it in a block, like this:

```
wait_for_ajax do
  expect(page).to have_content('Princess Bubblegum 1 started, 0 sent')
  expect(page).to have_content('Visitors (2) Lurkers (2) Leads (1)')
end
```

Reply:
Thanks for all the feedback. It. inspired me to do more research. I found this article: https://www.varvet.com/blog/why-wait_until-was-removed-from-capybara/ and decided to use assertions that would cause capybara to wait where necessary.

Reply:
LGTM!, let's see if this solves the issue.